livesync_citadel/indico_livesync_citadel/schemas.py

```diff
@@ -0,0 +1,266 @@
+# This file is part of the Indico plugins.
+# Copyright (C) 2002 - 2020 CERN
+#
+# The Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License;
+# see the LICENSE file for more details.
+
+from __future__ import unicode_literals
+
+import itertools
+
+from marshmallow import post_dump
+from tika import parser
+from webargs import fields
+
+from indico.core.db.sqlalchemy.links import LinkType
+from indico.core.db.sqlalchemy.principals import PrincipalType
+from indico.core.db.sqlalchemy.protection import ProtectionMode
+from indico.core.marshmallow import mm
+from indico.modules.attachments.models.attachments import Attachment, AttachmentType
+from indico.modules.events import Event
+from indico.modules.events.contributions.models.contributions import Contribution
+from indico.modules.events.contributions.models.subcontributions import SubContribution
+from indico.modules.events.notes.models.notes import EventNote
+from indico.web.flask.util import url_for
+
+
+def _get_location(obj):
+    if obj.venue_name and obj.room_name:
+        return '{}: {}'.format(obj.venue_name, obj.room_name)
+    elif obj.venue_name or obj.room_name:
+        return obj.venue_name or obj.room_name
+    else:
+        return ''
+
+
+def _get_identifiers(principal):
+    if principal.principal_type == PrincipalType.user:
+        # Instead of using the email this uses `User:ID`.
+        # Since the user can change the email this is better as
+        # it will ensure that only this given Indico user has access.
+        # If you want to stick with email, simply replace it with
+        # 'User:{}'.format(principal.email)
+        yield principal.identifier
+        yield principal.email
+    elif principal.principal_type == PrincipalType.event_role:
+        for user in principal.members:
+            yield user.identifier
+            yield principal.email
+    elif principal.is_group:
+        yield principal.identifier
+
+
+def _get_category_path(obj):
+    if isinstance(obj, Event):
+        event = obj
+    elif isinstance(obj, Attachment):
+        event = obj.folder.event
+    else:
+        event = obj.event
+    return event.category.chain_titles
+
+
+def _get_event_acl(event):
+    if event.effective_protection_mode == ProtectionMode.public:
+        acl = ['']
```

Comment:
i guess this empty string is intended?

Reply:
To be honest, I have no idea. This is something I didn't touch.

Reply:
I read the cern-search-rest-api code and apparently having an empty array will result in an Exception. I am not sure if that is intended or just a bug. Will discuss it with the search developers.

Reply:
Actually, `owner`, `update` and `delete` need to be set to a specific value which in our case (CERN) must correspond to an egroup. I am wondering if the plugin should have yet another setting which would be used to provide the default value for `delete`, `update`. Also access to the CERN Search (the management portal and place where we will be able to create API tokens) will be limited to a specific egroup.

Reply:
@ThiefMaster what do you think?

Reply:
if we need to reference some cern-specific group then yes, this should be configurable for sure